packages/flutter_tools/lib/src/runner/target_devices.dart

```diff
@@ -490,15 +510,40 @@ class TargetDevicesWithExtendedWirelessDeviceDiscovery extends TargetDevices {
         includeDevicesUnsupportedByProject: includeDevicesUnsupportedByProject,
         includeDisconnected: true,
       );
+
       if (devices.length == 1) {
         Device? matchedDevice = devices.first;
+
+        if (matchedDevice is IOSDevice && !matchedDevice.devModeEnabled) {
+          return _handleDisabledIosDevice(matchedDevice);
+        }
+
         if (!matchedDevice.isConnected && matchedDevice is IOSDevice) {
           matchedDevice = await _waitForIOSDeviceToConnect(matchedDevice);
         }
+
         if (matchedDevice != null && matchedDevice.isConnected) {
           return <Device>[matchedDevice];
         }
       }
+
+      if (devices.length > 1) {
+        final List<Device> disabledDevices = [];
```

Comment:
can use `where()` function to filter

Reply:
the idea is - for each device, if is indeed a disabled ios device, print the message that says "To use '
      "'$deviceName' for development, enable Developer Mode in Settings → Privacy & Security."
If, additionally, this is the last device to be checked, and all previous devices have also been disabled, then we print out all other available devices.
So I don't think I can actually use where() to filter it since I want to print something out each iteration

Reply:
info • Missing type annotation • packages/flutter_tools/lib/src/runner/target_devices.dart:531:46 • always_specify_types

```
final List<Device> disabledDevices = <Device>[];
```